src/Network/Network/NetworkVirtualAppliance/NetworkProfile/NewVirtualApplianceIpConfigurationCommand.cs

```diff
@@ -0,0 +1,57 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+using System.Linq;
+using Microsoft.Azure.Commands.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualApplianceIpConfiguration",
+        SupportsShouldProcess = true),
+        OutputType(typeof(PSVirtualApplianceIpConfiguration))]
+    public class NewVirtualApplianceIpConfigurationCommand : VirtualApplianceNetworkProfileBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = false,
+            HelpMessage = "The name of the IP configuration.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = false,
+            HelpMessage = "Indicates whether this IP configuration is the primary one.")]
+        [ValidateNotNullOrEmpty]
+        public bool Primary { get; set; }
```

Comment:
could you go through this guide: [Bool vs. SwitchParameter](https://github.com/Azure/azure-powershell/blob/ef3bd87e0aba10aa64c560b15e00ec7f10ec145b/documentation/development-docs/design-guidelines/parameter-best-practices.md#bool-vs-switchparameter) and double confirm it should be a bool or switchParameter?

Reply:
Ideally, we should use SwitchParameter, but we are using bool for other objects as well, so I'll take up the entire migration plan, I don't want to create inconsistent experiences for Customers.